spec/visual/samples/crystal

```diff
@@ -1,45 +1,687 @@
-lib LibC
-  WNOHANG = 0x00000001
-
-  @[ReturnsTwice]
-  fun fork : PidT
-  fun getpgid(pid : PidT) : PidT
-  fun kill(pid : PidT, signal : Int) : Int
-  fun getpid : PidT
-  fun getppid : PidT
-  fun exit(status : Int) : NoReturn
-
-  ifdef x86_64
-    alias ClockT = UInt64
-  else
-    alias ClockT = UInt32
+# Examples taken from http://crystal-lang.org/docs/
+# Copyright 2012-2018 Manas Technology Solutions.
+# Based on the sample in the Pygments project (github.com/pygments/pygments)
+
+## Constants
+
+LUCKY_NUMBERS     = [3, 7, 11]
+DOCUMENTATION_URL = "http://crystal-lang.org/docs"
+
+## Assignments
+
+i = 0
+items = Items.new
+channel = Channel(Int32).new(2)
+a = [] of Person
+hash = {} of Int32 => String
+nest = [1, ["b", [:c, ['d']]]]
+
+## Require statements
+
+require "http/server"
+
+## Blocks
+
+x = a.map { |f| f.name } # Error: can't infer block return type
+
+spawn do
+  channel.send(1)
+  channel.send(2)
+  channel.send(3)
+end
+
+3.times do |i|
+  puts channel.receive
+end
+
+server = HTTP::Server.new(8080) do |context|
+  context.response.content_type = "text/plain"
+  context.response.print "Hello world! The time is #{Time.now}"
+end
+
+## Loops
+
+while i < 10
+  proc = ->(x : Int32) do
+    spawn do
+      puts(x)
+    end
   end
+  proc.call(i)
+  i += 1
+end
 
-  SC_CLK_TCK = 3
+until some_condition
+  do_this
+end
+
+## Module definitions
 
-  struct Tms
-    utime : ClockT
-    stime : ClockT
-    cutime : ClockT
-    cstime : ClockT
+module HTTP
+  class RequestHandler
   end
+end
 
-  fun times(buffer : Tms*) : ClockT
-  fun sysconf(name : Int) : Long
+module Base64
+  extend self
+
+  def encode64(string)
+    # ...
+  end
 end
 
-class Process
-  def self.exit(status = 0)
-    LibC.exit(status)
+module Moo(T)
+  def t
+    T
   end
+end
+
+module Money
+  CURRENCIES = {
+    "EUR" => 1.0,
+    "ARS" => 10.55,
+    "USD" => 1.12,
+    "JPY" => 134.15,
+  }
 
-  def self.pid
-    LibC.getpid
+  class Amount
+    getter :currency, :value
+
+    def initialize(@currency, @value)
+    end
   end
 
-  def self.getpgid(pid : Int32)
-    ret = LibC.getpgid(pid)
-    raise Errno.new(ret) if ret < 0
-    ret
+  class CurrencyConversion
+    def initialize(@amount, @target_currency)
+    end
+
+    def amount
+      # implement conversion ...
+    end
   end
+Vend
```

Comment:
Is this intentional? Seems like it'd create a module that is never `end`ed but I am unsure of the spec goal

Reply:
Oops! Nope, didn't mean to have that there. Nice catch!